arbnode/dataposter/leveldb/leveldb.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021-2023, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package leveldb
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/offchainlabs/nitro/arbnode"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Storage implements leveldb based storage for batch poster.
+type Storage[Item any] struct {
+	db ethdb.Database
+}
+
+type Options struct {
+	File      string
+	Cache     int
+	Handles   int
+	Namespace string
+	ReadOnly  bool
+}
+
+func New[Item any](o *Options) (*Storage[Item], error) {
+	db, err := rawdb.NewLevelDBDatabase(o.File, o.Cache, o.Handles, o.Namespace, o.ReadOnly)
```

Comment:
Instead of opening a new db, we should use the existing rawdb (look at how block validator gets a copy of it)